ui/component/copyableStreamkey/view.jsx

```diff
@@ -0,0 +1,92 @@
+// @flow
+import * as ICONS from 'constants/icons';
+import { FormField } from 'component/common/form';
+import Button from 'component/button';
+import React, { useRef, Fragment } from 'react';
+
+type Props = {
+  copyable: string,
+  snackMessage: ?string,
+  doToast: ({ message: string }) => void,
+  primaryButton?: boolean,
+  name?: string,
+  onCopy?: (string) => string,
+  enableMask?: boolean,
+};
+
+export default function CopyableStreamkey(props: Props) {
+  const { copyable, doToast, snackMessage, primaryButton = false, name, onCopy, enableMask = true } = props;
+
+  const input = useRef();
+
+  function copyToClipboard() {
+    const topRef = input.current;
+    if (topRef[1].type === 'password') {
+      navigator.clipboard.writeText(topRef[1].defaultValue);
+    }
+    if (topRef[1].type === 'text') {
+      topRef[1].select();
+      if (onCopy) {
+        onCopy(topRef[1]);
+      }
+    }
+
+    document.execCommand('copy');
+  }
+
+  function checkMaskType() {
+    if (enableMask === true) {
+      return 'password';
+    }
+    if (enableMask === false) {
+      return 'text';
+    }
+  }
+  function showStreamkeyFunc() {
+    const topRef = input.current;
+    if (topRef[1].type === 'password') {
+      topRef[1].type = 'text';
+      topRef[0].innerText = 'Hide';
+    }
+    if (topRef[1].type === 'text') {
+      topRef[1].type = 'password';
+      topRef[0].innerText = 'Show';
+    }
+  }
```

Comment:
Issues: 
  - The "Show" button didn't work for me.
  - This is still not the React-way -- normally, we don't directly tweak the DOM.
  - The use of `<form>` is weird, as we are not submitting anything.

Take a look at how to use `useState`. 

A rough guide:
  - We probably don't need `<form>` and `type=password`
    - Without `password`, you don't have to change the way you perform the copy anymore.
  - Create a state, e.g. `showKey`
  - Replace `copyable` with asterix if state is hidden
     - e.g. `value={showKey ? copyable || '' : '***')`
       - Can make the asterix the same length if you want.

Reply:
Button is fixed.
I'm not using Dom, I'm using what was already in the file itself.
The <form> is being used to hold the main reference instead of a single FormField item.

Reply:
- The ref was originally used to highlight the text (i.e. `.select()`) when being copied, which we couldn't do via React.
- By default, we don't alter text directly that way.  Use states.
- `<form>` is not the right tag for that use-case.  A `<div>` would probably do.

## Merging
It works, and the changes are localized in 1 component, so good to go for now.  Thanks!